### ISSUE TITLE
How about a "WebMuxers" standard?

### ISSUE THREAD

**Issue by rafael2k** [dalecurtis/mediarecorder-muxer#1]
Instead of plugging the muxer API as an extension to MediaRecorder API, I think one option is to have a new standard API for W3C Muxers. I take as inspiration the WebM-Muxer API, as described here:

https://github.com/Vanilagy/webm-muxer

I think it would be a cleaner solution, which would support all the features for an audio / video / subtitles muxer, which is what we want when designing an audio or video editor, for example. I suggest starting with just webm as mandatory muxer, and than adding more as stack-holders demand it.

**Comment by dalecurtis** [comment]
Yeah a muxer/demuxer API would be better, I was trying to find a stop gap since my team won't have time to invest in such an API at this time though. I haven't thought through what shape the API should take, but that seems reasonable. We might want to ensure there's streaming support for muxing live content in an official api though.

**Comment by rafael2k** [comment]
I totally agree streaming support for muxing live content is desired. Webm-muxer, for example, currently does not support webm-streaming (but with small changes it could work, even if webm standard was not really intended for streaming...). Indeed, with AAC (adts format) WebCodec encoder I can already do live streaming from the browser using simple protocols, like icecast2 [1] or webcaster [2] (with liquidsoap server), which is pretty cool (and this is why I really love the AAC support in WebCodecs).

[1] https://icecast.org/
[2] https://github.com/webcast/webcaster
